erpnext/maintenance/doctype/maintenance_schedule/maintenance_schedule.js

```diff
@@ -120,3 +120,9 @@ cur_frm.cscript.generate_schedule = function(doc, cdt, cdn) {
 cur_frm.fields_dict.customer.get_query = function(doc,cdt,cdn) {
 	return { query: "erpnext.controllers.queries.customer_query" }
 }
+
+cur_frm.fields_dict['items'].grid.get_field('sales_person').get_query = function(doc, cdt, cdn) {
+	return{
+		filters: [['Sales Person', 'is_active', '=', 1]]
+	}
```

Comment:
If you use the fieldnames `enabled` or `disabled`, then the framework automatically does make these fields selectable, so you don't need to add `get_query` methods

Maybe use `enabled` and keep it checked by default